In [32]:
import pandas as pd
import pickle

In [33]:
with open('model_best_tunning.pkl', 'rb') as file_1:
  best_model_tunning = pickle.load(file_1)

In [34]:
inference_data = {
    "time": ["2025-08-09", "2025-08-13"],
    "latitude": [-8.65, -0.89],
    "longitude": [115.22, 119.87],
    "depth": [10.2, 35.4],
    "magType": ["mb", "mwr"],
    "nst": [45, 50],
    "gap": [65, 50],
    "dmin": [0.35, 0.20],
    "rms": [0.82, 0.65],
    "horizontalError": [0.7, 0.6],
    "depthError": [0.5, 0.4],
    "magError": [0.05, 0.07],
    "magNst": [40, 42],
    "status": ["reviewed", "reviewed"],
    "locationSource": ["us", "us"],
    "magSource": ["us", "us"],
    "date": ["14:23:00", "22:25:00"],
    "type": ["earthquake", "earthquake"],
    "location": ["Denpasar", "Palu"],
    "net": ["us", "us"],
    "id": ["us1234567", "us1234568"],
    }

df_inference = pd.DataFrame(inference_data)
df_inference

time  latitude  longitude  depth magType  nst  gap  dmin   rms  \
0  2025-08-09     -8.65     115.22   10.2      mb   45   65  0.35  0.82   
1  2025-08-13     -0.89     119.87   35.4     mwr   50   50  0.20  0.65   

   horizontalError  ...  magError  magNst    status locationSource magSource  \
0              0.7  ...      0.05      40  reviewed             us        us   
1              0.6  ...      0.07      42  reviewed             us        us   

       date        type  location net         id  
0  14:23:00  earthquake  Denpasar  us  us1234567  
1  22:25:00  earthquake      Palu  us  us1234568  

[2 rows x 21 columns]

In [35]:
df_inference.columns.unique()

Index(['time', 'latitude', 'longitude', 'depth', 'magType', 'nst', 'gap',
       'dmin', 'rms', 'horizontalError', 'depthError', 'magError', 'magNst',
       'status', 'locationSource', 'magSource', 'date', 'type', 'location',
       'net', 'id'],
      dtype='object')

In [36]:
# Prediksi langsung
y_pred = best_model_tunning.predict(df_inference)
print(y_pred)

[4.82907037 5.21758677]


Jadi, data 1 memprediksi gempa sebesar 4.8 SR dan data 2 memprediksi gempa sebesar 5.2 SR.